flocker/control/test/test_generations.py

```diff
@@ -0,0 +1,82 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Tests for ``flocker.node._generations``.
+"""
+
+from testtools.matchers import Equals, Is, Not
+
+from ...testtools import TestCase
+
+from ..testtools import deployment_strategy
+from .._generations import GenerationTracker
+from .._persistence import make_generation_hash
+
+
+class GenerationTrackerTests(TestCase):
+
+    def test_basic_use_works(self):
+        """
+        After inserting a bunch of deployments into a ``GenerationTracker`` in
+        sequence, the ``Diff`` returned from ``get_diff_from_hash_to_latest``
+        can be applied to convert each of the deployments to the latest
+        deployment.
+        """
+        deployments = list(deployment_strategy().example() for _ in xrange(5))
+        deployments[3] = deployments[1]
```

Comment:
:question: Yeah, you'll have to explain your strategy  / intent in this test too. I can't work it out :confused: